internal/mode/static/usage/job_worker.go

```diff
@@ -0,0 +1,62 @@
+package usage
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/config"
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/telemetry"
+)
+
+func CreateUsageJobWorker(
+	logger logr.Logger,
+	k8sClient client.Reader,
+	reporter Reporter,
+	cfg config.Config,
+) func(ctx context.Context) {
+	return func(ctx context.Context) {
+		nodeCount, err := telemetry.CollectNodeCount(ctx, k8sClient)
+		if err != nil {
+			logger.Error(err, "Failed to collect node count")
+		}
+
+		podCount, err := telemetry.CollectNGFReplicaCount(
```

Comment:
Because the cluster connector was cluster wide, this pod count should be all NGF replicas in a cluster - not just the pod count of this deployment. Unfortunately, the NIM endpoint doesn't give us a way to store counts from each deployment individually, so if there's more than one deployment in the cluster, they'll just keep overwriting each other.

Reply:
I think that was one of our talking points, and I thought we came to the conclusion to just report ourselves, even if it overwrites other reports. @mpstefan Is that accurate? Or was that purely relating to NIC vs NGF overwriting each other?

Reply:
Another point to add to this, is what value does that actually give us? There's no way for us to tell the difference between three replicas of one deployment, or one replica of three deployments.

Reply:
It's to support per-instance license counting - if a customer is using per-instance, each pod requires a license. So, if a customer has 2 deployments with 3 pods in each, they have a pod count of 6, but we will only ever report 3. Per instance licensing is legacy though, so maybe it no longer matters.

Reply:
@brianehlert May have insight on the requirements here.

Reply:
Per a conversation offline, we do want to report the total NGF pods across deployments. I'll use a best-effort approach to gather all replicasets with the `app.kubernetes.io/name=nginx-gateway-fabric` or `app.kubernetes.io/name:nginx-gateway` labels and return the count.